default/methods/source_control/file/file_browser.py

```diff
@@ -426,7 +474,7 @@ def view_file_list_web_route(project_string_id, username):
         return jsonify(file_list = output_file_list,
                        metadata = file_browser_instance.metadata), 200
 
-
+# TODO: Would like to take the `directory` param out of this class, but there's a lot of dependencies on it still.
```

Comment:
Doesnt make sense to pass a single directory to `File_Browser` since the queries support multiple directories. But there's a few areas in this class that depend on it, so left as is for now. Should I look at removing it?

Reply:
In general I wasnt sure how much we want to change `File_Browser` since its used in a few different places. We could just duplicate it for now so that we avoid breaking anything but start refactoring the new version of it?